### PR TITLE
fix(boxplot): hack boxplot into correct rendering using semiotic rc

### PR DESCRIPTION
**Pull request by anotherchrisberry**
Semiotic V2 is always returning `0` for `x` on data points and hover annotations, which causes very bad graph rendering.

These changes are very tactical, very brittle, I am not proud of them, and I hope they go away once Semiotic cuts a new release.

__Behavior with Semiotic V1.x__
Screen Shot 2021-04-14 at 2 15 44 PM

__With this PR__
Screen Shot 2021-04-14 at 2 15 54 PM

__Without this PR__
Screen Shot 2021-04-14 at 2 25 52 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "redux-observable": "^0.14.1",
     "reselect": "^3.0.1",
     "rxjs": "^5.4.2",
-    "semiotic": "^2.0.0-rc.5",
+    "semiotic": "2.0.0-rc.12",
```
Comment (anotherchrisberry) [inline package.json:66]: this makes no difference, I'm leaving it in the PR to show I tried

File: src/kayenta/report/detail/graph/semiotic/boxplot.tsx [hunk 2]
```diff
@@ -190,7 +192,7 @@ export default class BoxPlot extends React.Component<ISemioticChartProps, IBoxPl
           },
           className: 'boxplot-annotation',
         };
-        return <Annotation key={name} noteData={noteData} />;
+        return <Annotation key={`${d.group}: ${name}`} noteData={noteData} />;
```
Comment (anotherchrisberry) [inline src/kayenta/report/detail/graph/semiotic/boxplot.tsx:195]: unrelated, but this cuts out a few duplicate key warnings

**Comment by ranihorev** [comment]
Side notes: 
- I recommend updating the build command to `"build": "npm run clean && NODE_ENV=production rollup -c"`
- We need to upgrade the `@spinnaker/pluginsdk` to 0.0.26

**Comment by anotherchrisberry** [comment]
> Side notes:
> 
> * I recommend updating the build command to `"build": "npm run clean && NODE_ENV=production rollup -c"`
> * We need to upgrade the `@spinnaker/pluginsdk` to 0.0.26

done done

**Review by ranihorev — CHANGES_REQUESTED** [review]
Looks like it works only for a very specific screen size :)
I can try fixing it tomorrow if you want

image

**Comment by anotherchrisberry** [comment]
@ranihorev yeah, go for it! I played around with it some more but didn't have any luck.

I don't remember what the issue was that led to trying the RC version of semiotic, but there has been a more recent release of 1.x — maybe worth trying that if you don't have any luck, either.

**Comment by ranihorev** [comment]
I tried reverting back but I'm getting this error:
`Cannot read property 'contourRenderFn' of undefined`

**Pull request closed without merging**